src/platform/platform_worker.c

```diff
@@ -15,32 +15,53 @@
 #include "platform_worker.c.clog.h"
 #endif
 
+CXPLAT_RUNDOWN_REF CxPlatWorkerRundown;
+
+const uint32_t WorkerWakeEventPayload = CXPLAT_CQE_TYPE_WORKER_WAKE;
+const uint32_t WorkerUpdatePollEventPayload = CXPLAT_CQE_TYPE_WORKER_UPDATE_POLL;
+
 typedef struct QUIC_CACHEALIGN CXPLAT_WORKER {
 
     //
-    // Indicates if the worker is currently running.
+    // Flags to indicate what has been initialized.
     //
-    BOOLEAN Running;
+    BOOLEAN InitializedEventQ : 1;
+#ifdef CXPLAT_SQE_INIT
+    BOOLEAN InitializedShutdownSqe : 1;
+    BOOLEAN InitializedWakeSqe : 1;
+    BOOLEAN InitializedUpdatePollSqe : 1;
+#endif
+    BOOLEAN InitializedThread : 1;
+#ifdef QUIC_USE_EXECUTION_CONTEXTS
+    BOOLEAN InitializedECLock : 1;
+#endif
 
     //
-    // Event to wake the worker.
+    // Thread used to drive the worker.
     //
-    CXPLAT_EVENT WakeEvent;
+    CXPLAT_THREAD Thread;
 
     //
-    // Thread used to drive the worker.
+    // Event queue to drive execution.
     //
-    CXPLAT_THREAD Thread;
+    CXPLAT_EVENTQ EventQ;
 
+#ifdef CXPLAT_SQE
```

Comment:
Above you check if CXPLAT_SQE_INIT is defined for defining InitializedShutdownSqe etc, but here you check if CXPLAT_SQE is defined for having the ShutdownSqe etc fields. But then I see usage of ShutdownSqe gated on definition of CXPLAT_SQE_INIT down in e.g. CxPlatWorkersInit.

Is there not a single symbol you can gate this stuff on?

Reply:
No. Some platforms don't require it, some require it just for working space (no init required) and some require it to be initialized.